orchestrator/orchestrator.go

```diff
@@ -202,14 +202,14 @@ func (s *Orchestrator) outputToValue(output *process.Process_Node_Map_Output, wf
 			},
 		}, nil
 	case *process.Process_Node_Map_Output_Ref:
-		node, err := wf.FindNode(v.Ref.NodeKey)
-		if err != nil {
-			return nil, err
-		}
-		if node.GetTask() != nil {
-			return s.resolveInput(wf.Hash, exec, v.Ref.NodeKey, v.Ref.Key)
+		nodes := wf.FindNodes(func(n *process.Process_Node) bool {
```

Comment:
I correct the related issue https://github.com/mesg-foundation/engine/issues/1557
so `refKey` is optional
if not set, it the ref should be done on the outputs of the previous node (`return data.Fields[v.Ref.Key]`)

Reply:
we don't have a `Key` in Ref. And we set up that the ref key is not optional (same with path) or I missunderstood something